### DISCUSSION TITLE
[0916/102854.886141:FATAL:electron_main_delegate.cc(294)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

### DISCUSSION THREAD

**Discussion by sanyuanya** [NativePHP/laravel#209] [Q&A]
I used docker to build the php environment. After running php artisan native:serve, an error occurred:
[0916/102854.886141:FATAL:electron_main_delegate.cc(294)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

> Laravel Framework 10.23.1

> PHP 8.2.10 (cli) (built: Sep 7 2023 08:02:09) (NTS)

> Docker version 24.0.6, build ed223bc

**Comment by simonhamp** [comment]
Docker as a build environment isn't going to be supported at the moment.

You may find [some helpful guides](https://www.electron.build/multi-platform-build.html#docker) online if you search more generally for "electron build in docker", but my gut says it's going to be quite complex.

I highly recommend you [install PHP and Node](https://nativephp.com/docs/1/getting-started/installation#requirements) on the build machine and build directly from that machine's CLI.
